src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -111,38 +114,55 @@ export interface DiscoverSidebarResponsiveProps {
  */
 export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps) {
   const services = useDiscoverServices();
+  const { data, dataViews, core } = services;
   const isPlainRecord = useAppStateSelector(
     (state) => getRawRecordType(state.query) === RecordRawType.PLAIN
   );
   const { selectedDataView, onFieldEdited, onDataViewCreated } = props;
   const [fieldFilter, setFieldFilter] = useState(getDefaultFieldFilter());
   const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
-  /**
-   * fieldCounts are used to determine which fields are actually used in the given set of documents
-   */
-  const fieldCounts = useRef<Record<string, number> | null>(null);
-  if (fieldCounts.current === null) {
-    fieldCounts.current = calcFieldCounts(props.documents$.getValue().result!, selectedDataView);
-  }
+  const [allFields, setAllFields] = useState<DataViewField[] | null>(null);
+  const [fieldCounts, setFieldCounts] = useState<Record<string, number> | null>(null);
+  const dataViewFields = selectedDataView?.fields;
 
-  const [documentState, setDocumentState] = useState(props.documents$.getValue());
   useEffect(() => {
-    const subscription = props.documents$.subscribe((next) => {
-      if (next.fetchStatus !== documentState.fetchStatus) {
-        if (next.result) {
-          fieldCounts.current = calcFieldCounts(next.result, selectedDataView!);
-        }
-        setDocumentState({ ...documentState, ...next });
+    const subscription = props.documents$.subscribe((documentState) => {
+      if (documentState?.fetchStatus === FetchStatus.COMPLETE) {
+        setFieldCounts(calcFieldCounts(documentState.result));
+      } else {
+        setFieldCounts(null);
       }
     });
     return () => subscription.unsubscribe();
-  }, [props.documents$, selectedDataView, documentState, setDocumentState]);
+  }, [props.documents$, setAllFields, setFieldCounts]);
 
   useEffect(() => {
-    // when data view changes fieldCounts needs to be cleaned up to prevent displaying
-    // fields of the previous data view
-    fieldCounts.current = {};
-  }, [selectedDataView]);
+    setAllFields(getDataViewFieldList(selectedDataView, fieldCounts, isPlainRecord));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dataViewFields, fieldCounts, setAllFields, isPlainRecord]);
+
+  const query = useAppStateSelector((state) => state.query);
+  const filters = useAppStateSelector((state) => state.filters);
+  // TODO: we could save the current dateRange when documents change
+  const dateRange = data.query.timefilter.timefilter.getTime(); // TODO: is it correct to use the relative time range instead of absolute time range here? Currently, it helps to avoid unnecessary refetches.
```

Comment:
I'd suggest to align to Lens, doing an absolute time range request, also currently Discover field selector would miss new fields ingested, when an auto interval is set

Reply:
Switched to the absolute date range with rounding as @lukasolson [suggested](https://github.com/elastic/kibana/pull/144134) and it works better now :)

Thanks, Lukas and Matthias!